shared-libs/transitions/src/transitions/create_user_for_contacts.js

```diff
@@ -154,30 +154,49 @@ module.exports = {
       !contactType
       || !contactTypeUtils.isPersonType(contactType)
       || !doc.user_for_contact
-      || !doc.user_for_contact.replace
     ) {
       return false;
     }
 
-    return !!Object
-      .values(doc.user_for_contact.replace)
-      .find(({ status }) => status === USER_CREATION_STATUS.READY);
+    if (doc.user_for_contact.replace) {
```

Comment:
Good catch. It should still allow to add a user for the contact if there is no replacement going on.

```ts
if (doc.user_for_contact.replace && doc.user_for_contact.add) {
  const hasFinishedReplacing = Object
    .values(doc.user_for_contact.replace)
    .every(({ status }) => [USER_CREATION_STATUS.COMPLETE, USER_CREATION_STATUS.ERROR].includes(status));
  const isAddingUser = doc.user_for_contact.add.status === USER_CREATION_STATUS.READY;

  return hasFinishedReplacing && isAddingUser;
}
```

Reply:
Is there a particular reason we should prevent adding a new user _at the same time_ as we are replacing one?  I am not sure it would ever be possible to get into this situation, but it seems unnecessary to have extra logic to prevent that from happening....

Reply:
One thing to keep in mind here is the behavior of Sentinel when syncing multiple changes to a doc from an offline user.  Imagine you are offline and make changes A, B, and C to a doc. Then I come online and sync back to the server.  Sentinel will only submit the doc once (with the C revision) to the transitions.  So if a user is offline and triggers and _add_ and a _replace_ and then comes online and syncs, then I think we want the transition to be able to handle both these actions at the same time....